packages/labs/analyzer/src/lib/lit-element/events.ts

```diff
@@ -27,22 +31,26 @@ export const getEvents = (node: LitClassDeclaration) => {
         } else if (typeof comment === 'string') {
           const result = parseFiresTagComment(comment);
           if (result === undefined) {
-            // TODO(justinfagnani): report syntax error?
-            continue;
+            throw new DiagnosticsError(
+              tag,
+              'The @fires annotation was not in a recognized form. ' +
+                'Use `@fires event-name {Type} - Description`.'
+            );
           }
           const {name, type, description} = result;
-          // TODO(justinfagnani): how do we dereference the event type name
-          // into a TypeScript type? TypeScript will automatically do this for
-          // jsdoc tags it understands, but unfortunately @fires is not one of
-          // them.
           events.set(name, {
             name,
-            typeString: type,
+            type: type
```

Comment:
Per offline chat and to check understanding: this doesn't support generics like `CustomEvent<Foo>`. Will that fail here? And if not, can we make it fail with a diagnostic?

Let's make an issue for how to move forward on supporting this better. As discussed, we can either try to regex out a generic for the common case of `CustomEvent<Foo>` and assume that's in scope; or we can generate some TS with this code in scope and properly analyze it.

Reply:
Yes, `getTypeForConstructorName` will fail with on a generic with a diagnostic error because there won't be a symbol name matching that text. But I'm working on the latter now to make this just work.